#### Практическое задание. Урок 5. Проверка статистических гипотез. P-значения. Доверительные интервалы. A/B-тестирование

In [1]:
import numpy as np

##### 1. Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

По условию задачи нам известна дисперсия, поэтому используем z-критерий.

In [2]:
# Дано:
Xcp = M = 80
n=256
sigma = 16

Находим Z табличное для значения 0.975 (берем для a/2):

In [3]:
Za_2 = 1.96

In [4]:
lim1 = Xcp - Za_2 * sigma/n**(1/2)
lim1

78.04

In [5]:
lim1 = M + Za_2 * sigma/n**(1/2)
lim1

81.96

##### Ответ: доверительный интервал для оценки математического ожидания a с надежностью 0.95 находится в границах значений [78.04; 81.96]

##### 2. В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.

Сигма не известна, поэтому мы используем t-критерий.

In [6]:
X = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])

In [7]:
n = len(X)
n

10

In [8]:
M = Xcp = X.mean()
Xcp

6.590000000000001

In [9]:
sigma = X.std(ddof=1)
sigma

0.4508017549014448

t a/2 табличное находим в таблице для 95 доверительного интервала и степени свободы n-1:

In [10]:
t_a_2 = 2.262

In [11]:
lim1 = Xcp - t_a_2 * sigma/n**(1/2)
lim1

6.267538255912426

In [12]:
lim2 = Xcp + t_a_2 * sigma/n**(1/2)
lim2

6.912461744087575

##### Ответ: истинное значение величины X при 95% доверительном интервале находится в границах значений [6.267538255912426; 6.912461744087575]

##### 3. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

В соответствии с этапами проверки гипотез:

1. Сформулируем H0 и H1
H0: диаметр шариков = 17 мм, s=s0
H1: диаметр шариков = 17.5 мм, s>s0

2. Стандартное отклонение дано, оно равно 2, поскольку Дисперсия равна 4.
Следовательно, используем Z-критерий.

3. Уровень значимости дан, это односторонний критерий a = 0.05

In [13]:
#Определяем Z табличное по таблице, соответствующее z-критерию=95%.
Zt = 1.645

In [14]:
# Дано:
Xср = 17.5
s0 = 17
sigma = 2
n = 100
# Определяем Z расчетное по формуле:
Zp = (Xср-s0)/(sigma/n**(1/2))
Zp

2.5

##### Ответ: Гипотеза H0 не верна при заданном уровне значимости a=0.05, поскольку значение Z расчетного лежит за областью значений H0. Верна H1.

##### 4. Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

В соответствии с этапами проверки гипотез:

1. Сформулируем H0 и H1 H0: средний вес = 200 г, s=s0 H1: средний вес не равен 200 г, s!=s0

2. Стандартное отклонение  не дано. Следовательно, используем t-критерий.

3. Уровень значимости понятен из постановки задачи (s!=s0), это двухсторонний критерий a/2 = 0.01 (доверительная вероятность равна 99%)

In [15]:
X = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

In [16]:
#Определяем t табличное по таблице, соответствующее t-критерию=99% и степени свободы n-1
t_a_2 = 3.250

In [17]:
# Подсчет вводных данных для расчета
s0 = 200
n = 10
Xcp = X.mean()
sigma = X.std(ddof=1)

In [18]:
# Определяем Z расчетное по формуле:
tp = ((Xcp-s0)*n**(1/2))/sigma
tp

-1.0651074037450896

##### Ответ: Гипотеза H0 верна при заданном уровне значимости a=0.01, поскольку значение t расчетного лежит в области значений H0.